### ISSUE TITLE
Builder Repo Start - Killed code 137 ☠️☠️ - Step by Step with images

### ISSUE THREAD

**Issue by loveTSloveNET** [decentraland/builder#1678]
Hi Sir! :slightly_smiling_face:

**GOAL:**
Trying to start the Builder repo on my ubuntu 20.4 server.
https://github.com/decentraland/builder

**PROBLEM:**
Getting a error 137 back.

**STEP BY STEP:**
Here the commands i do:

**Step 1:** Navigate to the folder were i cloned the repo
```
cd /var/www/builder
```

**Step 2:** I install the decentraland compiler + run install command
```
npm i decentraland-rpc
npm install
```

**Step 3:** I rename the .env file
```
mv .env.example .env
```

**Step 4:** I open and edit the .env variables
```
nano .env 
```

**Step 5:**
![image](https://user-images.githubusercontent.com/24850331/142717776-49ea3117-2854-45a2-aaa8-93c5937c20c3.png)

**Step 6:** I start
```
npm start
```

**Step 7:** Error message code 137
![image](https://user-images.githubusercontent.com/24850331/142717780-68900d73-c2fa-4fd1-b131-23ee694092b9.png)

**Step 8:** I open the log
```
cd /root/.npm/_logs/2021-11-20T06-_21_05_501Z-debug.log
```

**Step 9:** The log is empty
![image](https://user-images.githubusercontent.com/24850331/142717787-fc1f158c-e8b0-4693-b226-a9b67a213f7b.png)

**QUESTIONS:**
Did i took for granted some step perhaps?
Any Hint?

Thanks so much!
:slightly_smiling_face: :pray:

Here my question on the forum:
[https://forum.decentraland.org/t/builder-repo-start-killed-code-137-step-by-step-with-images/2976]
(https://forum.decentraland.org/t/builder-repo-start-killed-code-137-step-by-step-with-images/2976)

Here my question on discord:
[https://discord.com/channels/417796904760639509/420166863776120832/911506864032841749](https://discord.com/channels/417796904760639509/420166863776120832/911506864032841749)

**Comment by loveTSloveNET** [comment]
Possible solution 1: (to see if i can make the client builder run on the server)
**Step 1:** back to my root
`cd /`

**Step 2:** give persmissions to npm
`sudo chown -R 65534:0 "root/.npm"`

**Step 3:** install decentraland
`npm i -g decentraland`

**Step 4:** initialize dcl
`dcl init`

**Step 5:** start dcl
`dcl start`

**Step 6:** check what ports are open
sudo lsof -i -P -n | grep LISTEN
![image](https://user-images.githubusercontent.com/24850331/142728952-f92daad4-c21a-43c1-80f1-46ac1385599f.png)

**Step 7:** tried listening the porst with nmap
```
sudo apt-get install nmap
sudo nmap -sTU -O 31.220.62.173
```
![image](https://user-images.githubusercontent.com/24850331/142729115-6371784b-df6b-42f9-916e-bd726974404f.png)

**Step 8:** typed on my browser the ip + double colon (:) + the port number
And got responses like:
"Page temporarly inactive"

**GOAL:** To type my server ip 
I was expecting to see the builder by typing my server ip https://31.220.62.173:3000
But for now am not making it work on the server
The SSL are activated, and i have done successful test with the https:// before

**Comment by loveTSloveNET** [comment]
Am successfully making it work on my local machine:
Here the terminal:
![image](https://user-images.githubusercontent.com/24850331/142729790-d148d638-0c23-4c2e-bf5f-2d05f3427168.png)

Here the browser
![image](https://user-images.githubusercontent.com/24850331/142729730-0ef0e1af-ea26-4c38-9f5b-645d75661360.png)

**Comment by loveTSloveNET** [comment]
Tried again from scratch on my Ubuntu 20.4 server:

**Step 1:** deleted the builder cloned folder from github
```
cd /var/www/
rm -rf builder
git clone https://github.com/Decentraland/builder.git
```

**Step 2:** installed dependencies
`npm install`

**Step 3:** started the webapp but same error
`npm start`
![image](https://user-images.githubusercontent.com/24850331/142730616-6b5a0f72-34d0-46d5-8dea-745b46c0ca03.png)

**Step 4:** open the log file
![image](https://user-images.githubusercontent.com/24850331/142730374-0af14957-4830-4a99-b0b5-1c3dce202d0a.png)

**Step 5:** on internet they reccomend to delete the modules and reinstall it again, but no change at all
```
npm cache clean --force
cd /var/www/builder
rm -rf node_modules
rm package-lock.json
```

**Comment by LautaroPetaccio** [comment]
Hi @loveTSloveNET ! We're not currently using Ubuntu to develop the builder, so we'll have to try it there to reproduce it, but there might be a couple of things that might help you build it:
- Use NodeJS 14
- Install all dependencies using `npm ci`

Would you mind trying that and letting us know if it worked?

**Comment by loveTSloveNET** [comment]
Thanks Sir!

Did a quick try reinstalling NodeJS but 14 instead, but had already installed another version and was tricky to reverse it.
Then had to format the server. Will give it a shot soon.

Was planning to use Devian next time. Will keep in mind to use NodeJS 14.
I order to abvoid small bumps on the road like this one I faced using Ubuntu, I was wondering, what OS are you using to develop it, to stick and use to that one aswell. Windows Server? Devian? I checked on the docker files but says just linux/64, any preferences over others about the VPS provider? Amazon? DigitalOcean?

Thanks! 😎 🙏
